gems/sorbet-runtime/lib/types/props/decorator.rb

```diff
@@ -333,6 +339,8 @@ def prop_defined(name, cls, rules={})
     end
     type_object = smart_coerce(type, enum: rules[:enum])
 
+    rules[:checked] ||= T::Private::RuntimeLevels.default_checked_level
```

Comment:
I think using `T::Private::RuntimeLevels.default_checked_level` is technically the right thing to do here, but there are some backwards compatibility concerns:
- Currently this is really `:always` since runtime prop checks are always evaluated. This would change behavior for people who have a different `default_checked_level` set
- By evaluating `T::Private::RuntimeLevels.default_checked_level`, we make it so calls to `T::Configuration.default_checked_level = ...` will fail. Since this is happening at the time any T::Struct is loaded into memory, that will likely break things for folks. See #6668 for a path to fix this for folks.

I think given the above, especially before we have a mechanism to allow folks to workaround load-order issues, we should probably change this to:
```suggestion
    rules[:checked] ||= :always
```

And then switch back to `T::Private::RuntimeLevels.default_checked_level` in a dedicated PR.

Reply:
NOTE: I wound up switching this back to `:always` to keep the behavior unchanged.